client/my-sites/comments/comment/comment-header.jsx

```diff
@@ -14,48 +14,58 @@ import Button from 'components/button';
 import CommentAuthor from 'my-sites/comments/comment/comment-author';
 import CommentAuthorMoreInfo from 'my-sites/comments/comment/comment-author-more-info';
 import FormCheckbox from 'components/forms/form-checkbox';
+import { getMinimumComment } from 'my-sites/comments/comment/utils';
 import { getSiteComment } from 'state/selectors';
 import { getSelectedSiteId } from 'state/ui/selectors';
 
-export const CommentHeader = ( {
-	commentId,
-	isBulkMode,
-	isEditMode,
-	isExpanded,
-	isSelected,
-	showAuthorMoreInfo,
-	toggleExpanded,
-} ) => (
-	<div className="comment__header">
-		{ isBulkMode && (
-			<label className="comment__bulk-select">
-				<FormCheckbox checked={ isSelected } onChange={ noop } />
-			</label>
-		) }
-
-		<CommentAuthor { ...{ commentId, isExpanded } } />
-
-		{ showAuthorMoreInfo && <CommentAuthorMoreInfo { ...{ commentId } } /> }
-
-		{ ! isBulkMode && (
-			<Button
-				borderless
-				className="comment__toggle-expanded"
-				disabled={ isEditMode }
-				onClick={ toggleExpanded }
-			>
-				<Gridicon icon="chevron-down" />
-			</Button>
-		) }
-	</div>
-);
+export class CommentHeader extends Component {
+	toggleSelected = () => this.props.toggleSelected( this.props.minimumComment );
```

Comment:
you could have kept this as a stateless component by returning a new `toggleSelected` on connect's `mergeProps`.
If you want to keep it as a `class`, I recommend it extends `PureComponent` instead.

Reply:
I'd rather keep it as a class, because more often than not all container components eventually need to be converted to class. 😄 
Updating to `PureComponent` though, good suggestion!